### PR TITLE
Reduced reliance performance plugins on NodeEngineImpl

### PR DESCRIPTION
**Pull request by pveentjer**
This cleanup is needed so that the same plugins can be used on the client side.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java [hunk 1]
```diff
@@ -55,34 +56,46 @@
      * @param hazelcastProperties array of {@link HazelcastProperty} to configure
      */
     protected void initProperties(Properties properties, HazelcastProperty[] hazelcastProperties) {
+        this.properties = properties;
         for (HazelcastProperty property : hazelcastProperties) {
```
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java]: Hmm, I know you haven't changed this, but Properties is mutable. Don't we want' to guard it?

Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java]: How do you want to guard it? There is no immutable wrapper available afaik.

Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java]: I think a user cannot change properties at runtime anyway, can't he?

Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java]: I'm working on a proper cleanup PR for the HazelcastProperties. There is some complexity here in the constructor and methods that need to be implemented that can be removed by simplification.

**Comment by Donnerbart** [comment]
:+1: (once green)

File: hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java (hunk 1 above)
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java]: A user should not be.

This is more a question about responsible programming. The question is if we need to protect the exposing collections and perhaps I should protect it. 

But @tombujok if you are fine with it; I can do this cleanup while dealing with the HazelcastProperties. I can get rid of the Properties completley and expose a keySet method which returns a immutable set and a getValue method. This way nobody can by accident modify a Properties-instance while accessing the HazelcastProperties

**Comment by tombujok** [comment]
:+1: let's rework the properties in other PR

**Pull request merged**